### DISCUSSION TITLE
Casting attributes that do not exist in the database is not supported?

### DISCUSSION THREAD

**Discussion by ngmy** [larastan/larastan#2154] [Q&A]
It appears that the cast is not properly recognized for attributes whose columns do not exist in the database.
Larastan says undefined variables.
This is a problem when doing value object casting, which is documented in the Laravel documentation.
https://laravel.com/docs/11.x/eloquent-mutators#value-object-casting

The versions I am using are as follows:
- PHP: 8.2.9
- Laravel: 11.36.1
- Larastan: 3.0.2

Example:
```php
/**
  * @return array{
  *     address: '\App\Casts\Address',
  * }
  */
protected function casts(): array
{
    return [
        'address' => Address::class,
    ];
}

$user->address; // App\ValueObjects\Address
\PHPStan\dumpType($user->address); // Access to an undefined property App\Models\User::$address.
```

Now I am using the following workaround:
```php
/**
 * @property \App\ValueObjects\Address $address
 */
```

**Comment by szepeviktor** [comment]
I think it is not about casting but generally the discovery of a property.
